runtimes/runtimes/lsp/router/util.ts

```diff
@@ -51,3 +51,13 @@ export function mergeObjects(obj1: any, obj2: any) {
     }
     return merged
 }
+
+export function findDuplicates<T>(array: T[]): T[] | undefined {
+    const seen = new Set<T>()
+    const dups = array
+        .filter(a => a !== undefined)
+        .filter(function (a) {
+            return seen.size === seen.add(a).size
```

Comment:
nit: this could be simplified with using `Set.has` method for checking seen element. I had to reread this few times to understand 😅

Reply:
it's shorter :), but I updated to make it more explicit